src/platform_impl/web/web_sys/event.rs

```diff
@@ -61,188 +62,59 @@ pub fn mouse_scroll_delta(event: &WheelEvent) -> Option<MouseScrollDelta> {
-        "Power" => VirtualKeyCode::Power,
-        "PrevTrack" => VirtualKeyCode::PrevTrack,
-        "AltRight" => VirtualKeyCode::RAlt,
-        "BracketRight" => VirtualKeyCode::RBracket,
-        "ControlRight" => VirtualKeyCode::RControl,
-        "ShiftRight" => VirtualKeyCode::RShift,
-        "MetaRight" => VirtualKeyCode::RWin,
-        "Slash" => VirtualKeyCode::Slash,
-        "Sleep" => VirtualKeyCode::Sleep,
-        "Stop" => VirtualKeyCode::Stop,
-        "NumpadSubtract" => VirtualKeyCode::NumpadSubtract,
-        "Sysrq" => VirtualKeyCode::Sysrq,
-        "Tab" => VirtualKeyCode::Tab,
-        "Underline" => VirtualKeyCode::Underline,
-        "Unlabeled" => VirtualKeyCode::Unlabeled,
-        "AudioVolumeDown" => VirtualKeyCode::VolumeDown,
-        "AudioVolumeUp" => VirtualKeyCode::VolumeUp,
-        "Wake" => VirtualKeyCode::Wake,
-        "WebBack" => VirtualKeyCode::WebBack,
-        "WebFavorites" => VirtualKeyCode::WebFavorites,
-        "WebForward" => VirtualKeyCode::WebForward,
-        "WebHome" => VirtualKeyCode::WebHome,
-        "WebRefresh" => VirtualKeyCode::WebRefresh,
-        "WebSearch" => VirtualKeyCode::WebSearch,
-        "WebStop" => VirtualKeyCode::WebStop,
-        "Yen" => VirtualKeyCode::Yen,
-        _ => return None,
-    })
+pub fn key(event: &KeyboardEvent) -> Key<'static> {
+    let key = event.key();
+    // TODO: Fix unbounded leak
+    let key = Box::leak(String::from(key).into_boxed_str());
+    Key::from_key_attribute_value(key)
 }
 
-pub fn keyboard_modifiers(event: &KeyboardEvent) -> ModifiersState {
-    let mut m = ModifiersState::empty();
-    m.set(ModifiersState::SHIFT, event.shift_key());
-    m.set(ModifiersState::CTRL, event.ctrl_key());
-    m.set(ModifiersState::ALT, event.alt_key());
-    m.set(ModifiersState::LOGO, event.meta_key());
-    m
+pub fn key_text(event: &KeyboardEvent) -> Option<&'static str> {
+    let key = event.key();
+    if let Key::Character(text) = Key::from_key_attribute_value(&key) {
+        // TODO: Fix unbounded leak
+        Some(Box::leak(String::from(text).into_boxed_str()))
+    } else {
+        None
+    }
 }
 
-pub fn codepoint(event: &KeyboardEvent) -> char {
-    // `event.key()` always returns a non-empty `String`. Therefore, this should
-    // never panic.
-    // https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key
-    event.key().chars().next().unwrap()
+pub fn key_location(event: &KeyboardEvent) -> KeyLocation {
+    let location = event.location();
+    // TODO: Use constants rather than integer literals?
+    match location {
+        0 => KeyLocation::Standard,
+        1 => KeyLocation::Left,
+        2 => KeyLocation::Right,
+        3 => KeyLocation::Numpad,
+        // TODO: Is this reasonable to do?
+        _ => KeyLocation::Standard,
+    }
+}
+
+pub fn key_repeat(event: &KeyboardEvent) -> bool {
+    event.repeat()
 }
+
+// TODO: What should be done about `KeyboardEvent.isComposing`?
```

Comment:
In that case, the keyboard event should probably be ignored for this PR. And there should be a `ReceivedImeText` emitted for [`compositionend`](https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionend_event).